runtime/src/main/java/io/quarkiverse/power/runtime/sensors/linux/rapl/IntelRAPLSensor.java

```diff
@@ -31,41 +34,70 @@ public IntelRAPLSensor() {
 
     private boolean checkAvailablity(String raplFileAsString) {
         final var raplFile = Path.of(raplFileAsString);
-        if (Files.exists(raplFile) && Files.isReadable(raplFile)) {
-            raplFiles.add(raplFile);
+        if (isReadable(raplFile)) {
+            // get metric name
+            final var nameFile = raplFile.resolveSibling("name");
+            if (isReadable(nameFile)) {
+                throw new IllegalStateException("No name associated with " + raplFileAsString);
+            }
+
+            try {
+                final var name = Files.readString(nameFile).trim();
+                raplFiles.put(name, mapRaplFile(raplFile));
+            } catch (IOException e) {
+                e.printStackTrace();
+                return false;
+            }
             return true;
         }
         return false;
     }
 
+    private static boolean isReadable(Path file) {
+        return Files.exists(file) && Files.isReadable(file);
+    }
+
     @Override
     public OngoingPowerMeasure<IntelRAPLMeasure> start(long duration, long frequency, Writer out) throws Exception {
-        return new OngoingPowerMeasure<>(new IntelRAPLMeasure(extractPowerMeasure()));
+        this.frequency = (double) frequency / 1000;
+        IntelRAPLMeasure measure = new IntelRAPLMeasure();
+        update(measure);
+        return new OngoingPowerMeasure<>(measure);
     }
 
-    @Override
-    public void update(OngoingPowerMeasure<IntelRAPLMeasure> ongoingMeasure, Writer out) {
-        ongoingMeasure.addCPU(extractPowerMeasure());
+    private void update(IntelRAPLMeasure measure) {
+        double cpuShare = PowerMeasurer.instance().cpuShareOfJVMProcess();
+        raplFiles.forEach((name, buffer) -> measure.updateValue(name, extractPowerMeasure(buffer), cpuShare, frequency));
     }
 
-    private double extractPowerMeasure() {
-        return extractPowerMeasure(raplFiles.stream().map(IntelRAPLSensor::readLongFromFile), PowerMeasurer.instance());
+    @Override
+    public void update(OngoingPowerMeasure<IntelRAPLMeasure> ongoingMeasure, Writer out) {
+       update(ongoingMeasure.sensorMeasure());
     }
 
-    private static long readLongFromFile(Path f) {
-        try {
-            return Long.parseLong(Files.readString(f).trim());
-        } catch (IOException e) {
-            throw new RuntimeException(e);
+    private static ByteBuffer mapRaplFile(Path file) throws IOException {
+        try (var channel = FileChannel.open(file, StandardOpenOption.READ)) {
+            ByteBuffer buffer = ByteBuffer.allocate((int) channel.size());
```

Comment:
The ByteBuffer here should be better pooled somewhere

Reply:
My brain is fried, I guess, can't get it to work now, will get back to it on Monday.